### PR TITLE
VecMem Update, main branch (2023.03.29.)

### PR DESCRIPTION
**Pull request by krasznaa**
Updated the project to use [vecmem-0.24.0](https://github.com/acts-project/vecmem/releases/tag/v0.24.0).

A bit surprisingly none of the tests in this project use vector buffers, so the code didn't actually need to be updated...

### CODE REVIEW

**Review by beomki-yeo — APPROVED** [review]
:+1:

**Comment by krasznaa** [comment]
It was only a technical issue, but we should still wait for the CI to succeed before we would merge anything...

**Pull request merged**
